test/unit/expirationd_stats_test.lua

```diff
@@ -25,22 +34,54 @@ function g.test_stats_basic()
     task:kill()
 end
 
-function g.test_stats_expired_count()
+function g.test_stats_expired_count(cg)
     helpers.iteration_result = {}
-    g.hash:insert({1, "a"})
-    g.hash:insert({2, "b"})
-    g.hash:insert({3, "c"})
+    cg.space:insert({1, "a"})
+    cg.space:insert({2, "b"})
+    cg.space:insert({3, "c"})
 
-    local task = expirationd.start("stats_expired_count", g.hash.id, helpers.is_expired_debug)
-    helpers.retrying({}, function()
-        t.assert_equals(helpers.iteration_result, {
+    local expected = {
+        {3, "c"},
+        {2, "b"},
+        {1, "a"},
+    }
+    if cg.params.index_type == 'TREE' then
+        expected = {
+            {1, "a"},
+            {2, "b"},
             {3, "c"},
+        }
+    end
+
+    expirationd.start("stats_expired_count", cg.space.id, helpers.is_expired_debug)
+    helpers.retrying({}, function()
+        t.assert_covers(helpers.iteration_result, expected)
+    end)
+
+    helpers.iteration_result = {}
+    cg.space:insert({1, "a"})
+    cg.space:insert({2, "b"})
+    cg.space:insert({3, "c"})
+
+    local iteration_result = {
+        {3, "c"},
+        {2, "b"},
+        {1, "a"},
+    }
+    if cg.params.index_type == 'TREE' then
```

Comment:
It requires a comment. Or probably more explicit conditiobn:
```
local var
if cond1 then
elseif cond2 then
else error('unreachable')
end
```

Reply:
Updated.

```lua
diff --git a/test/unit/expirationd_stats_test.lua b/test/unit/expirationd_stats_test.lua
index e8b504a..6c7156c 100644
--- a/test/unit/expirationd_stats_test.lua
+++ b/test/unit/expirationd_stats_test.lua
@@ -40,22 +40,26 @@ function g.test_stats_expired_count(cg)
     cg.space:insert({2, "b"})
     cg.space:insert({3, "c"})
 
-    local expected = {
-        {3, "c"},
-        {2, "b"},
-        {1, "a"},
-    }
+    local iteration_result
     if cg.params.index_type == 'TREE' then
-        expected = {
+        iteration_result = {
             {1, "a"},
             {2, "b"},
             {3, "c"},
         }
+    else if cg.params.index_type == 'HASH' then
+        iteration_result = {
+            {3, "c"},
+            {2, "b"},
+            {1, "a"},
+        }
+    else
+        error('Expected result is undefined.')
     end
 
     expirationd.start("stats_expired_count", cg.space.id, helpers.is_expired_debug)
     helpers.retrying({}, function()
-        t.assert_covers(helpers.iteration_result, expected)
+        t.assert_covers(helpers.iteration_result, iteration_result)
     end)
 
     helpers.iteration_result = {}
@@ -63,19 +67,6 @@ function g.test_stats_expired_count(cg)
     cg.space:insert({2, "b"})
     cg.space:insert({3, "c"})
 
-    local iteration_result = {
-        {3, "c"},
-        {2, "b"},
-        {1, "a"},
-    }
-    if cg.params.index_type == 'TREE' then
-        iteration_result = {
-            {1, "a"},
-            {2, "b"},
-            {3, "c"},
-        }
-    end
-
     local task = expirationd.start("stats_expired_count", cg.space.id, helpers.is_expired_debug)
     helpers.retrying({}, function()
         t.assert_equals(helpers.iteration_result, iteration_result)
```